src/analyses/variable-sensitivity/abstract_object.h

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************\
+
+ Module: analyses variable-sensitivity
+
+ Author: Thomas Kiley, thomas.kiley@diffblue.com
+
+\*******************************************************************/
+#ifndef CPROVER_ANALYSES_VARIABLE_SENSITIVITY_ABSTRACT_OBJECT_H
+#define CPROVER_ANALYSES_VARIABLE_SENSITIVITY_ABSTRACT_OBJECT_H
+
+class typet;
+class constant_exprt;
+class abstract_environmentt;
+class namespacet;
+
+#include <util/expr.h>
+#include <memory>
+
+
+#define CLONE \
+  virtual abstract_objectt* clone() const \
+  { \
+    typedef std::remove_const<std::remove_reference<decltype(*this)>::type \
+      >::type current_typet; \
```

Comment:
Yes, this is valid C++11. Yet adding a simple `typedef some_classt selft;` to each of those classes and then doing `return new selft(*this);` might make for much simpler code. [But I'm yet to find out whether all this is really necessary at all.]